### ISSUE TITLE
importlib.reload(module) broken for python versions > 3.4

### ISSUE THREAD

**Issue by skamsie** [neovim/pynvim#285]
If you are trying to use the `importlib.reload` from python version higher than 3.4, you get an error that comes from neovim's library. Works fine for lower versions.

```
Error detected while processing function provider#python3#Call:
line   18:
Traceback (most recent call last):
  File "<string>", line 6, in <module>
  File "/Users/.../.pyenv/versions/3.6.2/lib/python3.6/imp.py", line 314, in reload
    return importlib.reload(module)
  File "/Users/.../.pyenv/versions/3.6.2/lib/python3.6/importlib/__init__.py", line 165, in reload
    spec = module.__spec__ = _bootstrap._find_spec(name, pkgpath, target)
  File "<frozen importlib._bootstrap>", line 885, in _find_spec
  File "<frozen importlib._bootstrap_external>", line 1157, in find_spec
  File "<frozen importlib._bootstrap_external>", line 1129, in _get_spec
  File "/Users/.../.pyenv/versions/neovim362/lib/python3.6/site-packages/neovim/plugin/script_host.py", line 226, in find_spec
    return PathFinder.find_spec(fullname, path or _get_paths(), target)
  File "<frozen importlib._bootstrap_external>", line 1157, in find_spec
  File "<frozen importlib._bootstrap_external>", line 1123, in _get_spec
TypeError: 'module' object is not iterable
```

To reproduce:

1. setup neovim with python > 3.4 only (I did it following [this tutorial](https://github.com/zchee/deoplete-jedi/wiki/Setting-up-Python-for-Neovim))
2. use a plugin that uses `imp.reload` or `importlib.reload`, I used https://github.com/ryanss/vim-hackernews
3. Use the command `:HackerNews` once, and if you use it the second time in the same nvim session you get the error above. It can be reproduced with any plugin that uses `importlib.reload`. Also this does not reproduce with vim with python3.6

I am using latest neovim:

neovim version: NVIM v0.2.2
neovim python package: neovim (0.2.0)

**EDIT** the part of the plugin that uses `reload` can be seen here: https://github.com/ryanss/vim-hackernews/blob/master/ftplugin/hackernews.vim#L21-L32

**Comment by skamsie** [comment]
Seems to work fine if I comment out this method locally 🤷‍♂️ ... I assume there is more context behind why this check is needed though.

https://github.com/neovim/python-client/blob/master/neovim/plugin/script_host.py#L223-L226

**Comment by bfredl** [comment]
Does "work fine" include the functionality this code provides, namely importing from `{rtp}/python3` directories?

**Comment by skamsie** [comment]
@bfredl I am not sure how to test that, but by 'work fine' I meant I get no error...

**Comment by bfredl** [comment]
you could add `python3/thetestfile.py` to `~/.config/nvim` or whereever your config is and try `py3 import thetestfile`

**Comment by skamsie** [comment]
Yes I can confirm that commenting out that part of code results in the expected functionality. My naive way of testing:

1. I have commented out [this part of the hackernews code locally](https://github.com/ryanss/vim-hackernews/blob/master/ftplugin/hackernews.vim#L11-L18) to force only using the python3 version because Neovim seems to force `g:python_host_prog` even if you don't set it.
2. I have commented out this method as in my previous comment: https://github.com/neovim/python-client/blob/master/neovim/plugin/script_host.py#L223-L226
3. I started nvim and used the command `:HackerNews` and the content of hackernews was loaded into the buffer. In a different tab, I opened [this file] (https://github.com/ryanss/vim-hackernews/blob/master/ftplugin/hackernews.py#L104) and modified it so it only prints the title
4. I used the `:HackerNews` command again and the buffer was only populated with the title which means the module was successfully reloaded

**Disclaimer** I am not trying to advertise the hackernews plugin in any way. It's just an example. This bug reproduces only for modules loaded through plugins and even if I have something similar locally it would be a pain to explain how it's set up.
